extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java

```diff
@@ -126,6 +124,10 @@ ServiceStartBuildItem finalizeRouter(
                 listOfFilters, vertx.getVertx(), router.getRouter(), httpBuildTimeConfig.rootPath, launchMode.getLaunchMode());
 
         boolean startVirtual = requireVirtual.isPresent() || httpConfiguration.virtual;
+        if (requireVirtual.isPresent() || httpConfiguration.virtual) {
```

Comment:
Can you revert this change and the relevant removal please?

Reply:
The `HttpConfiguration` class is a runtime configuration, we are trying to access a value from a normal build step (build time), that's what motivated to move that part here (inside a RUNTIME recorder).  Is this okay?

Reply:
The fact that the build step is also a recorder doesn't affect the access of build / runtime time configuration.
However your latest addition does make things a little more clear so let's go with it. I'll just add one comment about the name.